libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -1317,10 +1350,34 @@ void ExpressionCompiler::appendArithmeticOperatorCode(Token::Value _operator, Ty
 		m_context << Instruction::SUB;
 		break;
 	case Token::Mul:
-		m_context << Instruction::MUL;
+		if (c_isFractional)
+		{
+			//multiply, then shift right...this is your fraction.
```

Comment:
I used these as my notes. http://x86asm.net/articles/fixed-point-arithmetic-and-tricks/